pkg/controller/node/nodecontroller.go

```diff
@@ -324,6 +330,19 @@ func (nc *NodeController) Run(period time.Duration) {
 	go wait.Until(nc.cleanupOrphanedPods, 30*time.Second, wait.NeverStop)
 }
 
+func (nc *NodeController) filterOutServiceRange() {
+	if !nc.clusterCIDR.Contains(nc.serviceCIDR.IP) && !nc.serviceCIDR.Contains(nc.clusterCIDR.IP) {
+		return
+	}
+
+	for i := 0; i< nc.cidrAllocator.MaxCIDRs(); i++ {
```

Comment:
This will be extremely inefficient. CIDR range is, in practice an interval of integers, and you're using a bitmask to keep the occupied CIDRs, so you can mark whole service range in a single call.